### PR TITLE
Calldata reader 

### PR DESCRIPTION
**Pull request by saifalkatout**
Implementation of CalldataReader. An instance of CalldataReader can be created using ctx.calldata_reader().

### CODE REVIEW

File: crates/library/std/src/context.fe [hunk 1]
```diff
@@ -19,6 +19,60 @@ pub trait Emittable {
...
+        self.cur_offset += 32
+        assert self.cur_offset <= self.len
+        return self.cur_offset
+    }
+
+    fn read_n(mut self, len: u256) -> u256 {
+        unsafe {
+            let value: u256 = evm::call_data_load(offset: self.cur_offset)
+            self.advance(len)
+            if len < 32 {
```
Comment (saifalkatout) [inline crates/library/std/src/context.fe:44]: The if statement here is somewhat costly, is the increased readability/abstration of making all the read_ functions use read_n worth it ??

Comment (g-r-a-n-t) [reply crates/library/std/src/context.fe:44]: the `read_u256` function could just use `call_data_load` itself and then this if statement can be removed

Comment (saifalkatout) [reply crates/library/std/src/context.fe:44]: Wouldn't that be redundant and reduce readability ??

**Review by g-r-a-n-t — APPROVED** [review]
nice work

**Pull request merged**